src/ngraph/builder/norm.cpp

```diff
@@ -36,125 +37,147 @@ namespace ngraph
     {
         namespace detail
         {
-            shared_ptr<Node> lp_norm(const Output<Node>& value,
-                                     size_t p_norm,
-                                     const AxisSet& reduction_axes,
-                                     float bias)
+            namespace opset1
             {
-                // In general "entrywise" lp-norm for matrix `A` is defined as following double sum:
-                // ||A||_p = ||vec(A)||_p = [sum_{i=1}^m sum_{j=1}^n abs(a_{i,j})^p]^{1/p}
-                shared_ptr<Node> abs_values{make_shared<op::Abs>(value)};
-                shared_ptr<Node> p_node = op::Constant::create(
-                    value.get_element_type(),
-                    value.get_shape(),
-                    vector<float>(shape_size(value.get_shape()), static_cast<float>(p_norm)));
-
-                // Get inner part of equation: abs_values^p_node, then sum over reduction_axes.
-                shared_ptr<Node> values{make_shared<op::Power>(abs_values, p_node)};
-                values = make_shared<op::Sum>(values, reduction_axes);
-
-                shared_ptr<Node> bias_node{
-                    op::Constant::create(values->get_element_type(),
-                                         values->get_shape(),
-                                         vector<float>(shape_size(values->get_shape()), bias))};
-
-                values = values + bias_node;
-
-                // Get outer part of equation: raise values to 1/p_norm exponent.
-                shared_ptr<Node> inv_p_node = op::Constant::create(
-                    values->get_element_type(),
-                    values->get_shape(),
-                    vector<float>(shape_size(values->get_shape()), 1.f / p_norm));
-
-                return {make_shared<op::Power>(values, inv_p_node)
-                            ->add_provenance_group_members_above({value})};
+                shared_ptr<Node> lp_norm(const Output<Node>& value,
+                                         size_t p_norm,
+                                         const AxisSet& reduction_axes,
+                                         float bias)
+                {
+                    // In general "entrywise" lp-norm for matrix `A` is defined as following double
+                    // sum:
+                    // ||A||_p = ||vec(A)||_p = [sum_{i=1}^m sum_{j=1}^n abs(a_{i,j})^p]^{1/p}
+                    shared_ptr<Node> abs_values{make_shared<ngraph::opset1::Abs>(value)};
+                    shared_ptr<Node> p_node = ngraph::opset1::Constant::create(
```

Comment:
We can create here just a scalar with `p_norm` value and later relay on NumPy autobroadcasting mechanism in `Power` op.

Reply:
done